src/pages/Strawpoll.tsx

```diff
@@ -97,8 +97,10 @@ export function DeleteStrawpollModal(props: ModalProps) {
 export default function Strawpoll(props: StrawpollProps) {
   const { committeeID, strawpollID } = props.match.params;
   const strawpollFref = getStrawpollRef(committeeID, strawpollID)
-  const [value, loading] = useObject(strawpollFref);
-  const [user] = useAuthState(firebase.auth());
+  // TODO: Bandaid - Another firebase type mismatch, it's expecting type Query but not entirely sure why
+  const [value, loading] = useObject(strawpollFref as any);
```

Comment:
I think this might need to be `useObjectVal`. `useObject` gives you a `DataSnapshot` in its `0th` param.

Reply:
Hmm, `useObjectVal` still has the same type error where it's expecting type `Query` as an argument but getting type `Reference`

Reply:
```ts
export declare const useObject: (query?: Query | null | undefined) => ObjectHook;
export declare const useObjectVal: <T, KeyField extends string = "", RefField extends string = "">(query?: Query | null | undefined, options?: ValOptions<T> | undefined) => ObjectValHook<T, KeyField, RefField>;
```

```
Argument of type 'Reference' is not assignable to parameter of type 'Query'.
The types of 'ref.isEqual' are incompatible between these types.
Type '(other: firebase.default.database.Query | null) => boolean' is not assignable to type '(other: import("/Users/geordie/Documents/Muncoordinated-2/node_modules/@firebase/database/dist/public").Query | null) => boolean'.
Types of parameters 'other' and 'other' are incompatible.
Type 'import("/Users/geordie/Documents/Muncoordinated-2/node_modules/@firebase/database/dist/public").Query | null' is not assignable to type 'firebase.default.database.Query | null'.
Type 'Query' is missing the following properties from type 'Query': endAt, endBefore, equalTo, limitToFirst, and 11 more.
```

Reply:
ah yep `useObjectVal` seems to work with the redesigned strawpollFref but I'm not sure how to use the `child` constructors. Could you provide an example? Here is my feeble attempt

Original
```ts
  const addOption = () => {
    strawpollFref.child('options').push(DEFAULT_STRAWPOLL_OPTION);
  }
```

New version something like this??
```ts
  const addOption = async () => {
    const strawpollFref = getStrawpollRef(committeeID, strawpollID);
    const newOptionRef = push(strawpollFref);
    await set(newOptionRef, DEFAULT_STRAWPOLL_OPTION);
  }
```

Reply:
Something like this?

```ts
  const addOption = async () => {
    const newOptionRef = push(strawpollFref);
    await set(newOptionRef, DEFAULT_STRAWPOLL_OPTION);
  }
```

Reply:
that looks right!

Reply:
Great, that makes it clear, thanks!

Do you think there Is any way we can get around these 2 patches without using the firebase modular API? 

The problem is, using the new modular API means a `DatabaseReference` object gets returned as opposed to a `Reference` object, this is what triggers the `Strawpoll` re-write but the ref is passed from `Strawpoll` to the handlers so there's a type mismatch, if the handler is updated, then everything goes buck wild and it seems like the entire project will be needed to be moved off the compat API.

`strawpollFref` of new type passed from here:
```ts
        <Input
          value={strawpoll ? strawpoll.question : ''}
          onChange={fieldHandler<StrawpollData>(strawpollFref, 'question')}
          fluid
          placeholder="Type your question here"
        />
```

To here:
```ts
export function fieldHandler<T>
(fref: firebase.database.Reference, field: keyof T) {
  return (e: React.FormEvent<HTMLInputElement>) =>
    fref.child(field.toString()).set(e.currentTarget.value);
}
```

and the chain reaction continues, until i imagine everything will be converted.

Reply:
I don't mind doing it if I have to, but if it can be avoided that might be nice.

Reply:
I think this motivates a second `fieldHandler` using the new `DatabaseReference`. Perhaps a `modularFieldHandler`. That should prevent the cascade.

Reply:
Oh yeah great idea